Sources/PostgresNIO/Data/PostgresData+Range.swift

```diff
@@ -0,0 +1,244 @@
+        return "\(opener)\(lowerBoundString),\(upperBoundString),\(closer)"
+    }
+}
+
+// MARK: PostgresData
+
+extension PostgresData {
+    public func range<B>() -> PostgresRange<B>?
+        where B: PostgresRangeBound
+    {
+        guard var value: ByteBuffer = self.value else {
+            return nil
+        }
+
+        return PostgresRange<B>(
+            from: &value,
+            format: self.formatCode
+        )
+    }
+
+    public init<B>(range: PostgresRange<B>)
+    {
+        var buffer = ByteBufferAllocator().buffer(capacity: 0)
+        range.encode(into: &buffer)
+
+        self.init(
+            type: B.rangeType,
+            typeModifier: nil,
+            formatCode: .binary,
+            value: buffer
+        )
+    }
+}
+
+// MARK: PostgresDataConvertible conformance
+
+extension PostgresRange: PostgresDataConvertible where B: PostgresRangeBound {
+    public static var postgresDataType: PostgresDataType { return B.rangeType }
+    public init?(postgresData: PostgresData) {
+        guard var value = postgresData.value else {
+            return nil
+        }
+        self.init(from: &value, format: .binary)
+    }
+    public var postgresData: PostgresData? {
+        return PostgresData(range: self)
+    }
+}
+
+extension RangeExpression where Bound: PostgresRangeBound {
+    public static var postgresDataType: PostgresDataType {
+        return Bound.postgresDataType
+    }
+}
+
+extension Range: PostgresDataConvertible where Bound: PostgresRangeBound {
+    public init?(postgresData: PostgresData) {
+        guard let postgresRange: PostgresRange<Bound> = postgresData.range(),
+            let lowerBound: Bound = postgresRange.lowerBound,
+            let upperBound: Bound = postgresRange.upperBound,
+            postgresRange.isLowerBoundInclusive,
+            !postgresRange.isUpperBoundInclusive
+        else {
+            return nil
+        }
+
+        self = lowerBound..<upperBound
+    }
+
+    public var postgresData: PostgresData? {
+        return PostgresData(range: PostgresRange<Bound>(range: self))
+    }
+}
+
+extension ClosedRange: PostgresDataConvertible where Bound: PostgresRangeBound {
+    public init?(postgresData: PostgresData) {
+        guard let postgresRange: PostgresRange<Bound> = postgresData.range(),
+            let lowerBound: Bound = postgresRange.lowerBound,
+            var upperBound: Bound = postgresRange.upperBound,
+            postgresRange.isLowerBoundInclusive
+        else {
+            return nil
+        }
+
+        if !postgresRange.isUpperBoundInclusive,
+            let steppedDownUpperBound = upperBound.stepDown?()
+        {
+            upperBound = steppedDownUpperBound
+        }
+
+        self = lowerBound...upperBound
+    }
+
+    public var postgresData: PostgresData? {
+        return PostgresData(range: PostgresRange<Bound>.init(closedRange: self))
+    }
+}
```

Comment:
Would it be okay for you if we don't add conformance to `PostgresDataConvertible`? We have already started deprecating this API in #313.

Reply:
See new commit https://github.com/vapor/postgres-nio/pull/330/commits/52f1d292197f6043bb242855cd6a3a084b864046